src/model/types/index.ts

```diff
@@ -556,6 +556,11 @@ export type AmountPreference = {
 	count: number;
 };
 
+export type Preferences = {
+	sendPreference: Array<AmountPreference>;
+	keepPreference?: Array<AmountPreference>;
+};
```

Comment:
This is an awesome PR!

If we do introduce breaking changed is on how the preference is passed, what do you guys think of getting rid of the (rather cumbersome?) format for the preference as `sendPreference: [{ amount: 1, count: 3 }]` and instead use a pure array of numbers a la `sendPreference: [1, 1, 1]`? I think I would prefer that.

Reply:
Probably a good idea. I think the original motive behind it was noble (not having an array with a billion useless 1s in it) but this is more practical.

Reply:
Yes that was exactly the thought behind it, but I came around to the idea of simpler data structures

Reply:
Is this the matter for another PR, thought? Or shall we do it here?

Reply:
we can do it in a different PR